services/slack/download_test.go

```diff
@@ -0,0 +1,198 @@
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData, 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
+
+	t.Run("successful re-download, empty file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, []byte{}, 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/no_resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
+
+	t.Run("successful re-download, tiny file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData[:8], 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/no_resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
+
+	t.Run("successful re-download, half file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData[:1024*512], 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/no_resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
+
+	t.Run("successful re-download, full file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData, 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/no_resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
+
+	t.Run("unsuccessful resume, tiny file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData[:8], 0660))
+
+		require.Error(t, downloadInto(fileName, srv.URL+"/wrong_resume", int64(len(mockData))))
+	})
+
+	t.Run("unsuccessful resume, half file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData[:1024*512], 0660))
+
+		require.Error(t, downloadInto(fileName, srv.URL+"/wrong_resume", int64(len(mockData))))
+	})
+
+	t.Run("successful resume from wrong file with an already downloaded file", func(t *testing.T) {
+		fileName := filepath.Join(os.TempDir(), "download-test")
+		defer os.Remove(fileName)
+		require.NoError(t, os.WriteFile(fileName, mockData, 0660))
+
+		require.NoError(t, downloadInto(fileName, srv.URL+"/wrong_resume", int64(len(mockData))))
+		tempFile, _ := os.ReadFile(fileName)
+		require.Equal(t, mockData, tempFile)
+	})
```

Comment:
Interesting case. Should we error there? There is no overlap between wrong file and already downloaded file. Can we make sure the one downloaded is the right one? In any case a super edge case, so up to you :)

Reply:
Ah cool, then let's leave it as it is. Thanks for the context :)